sim/core/cast.go

```diff
@@ -65,6 +61,10 @@ type Cast struct {
 	AfterCastDelay time.Duration
 }
 
+func (c *Cast) empty() bool {
+	return c.Cost == 0 && c.GCD == 0 && c.CastTime == 0 && c.ChannelTime == 0 && c.AfterCastDelay == 0
```

Comment:
You can just do `*c == Cast{}`

Reply:
Nope :), this would require an extra `var e Cast` line. Can do that, of course, safer if `Cast` ever changes. Alternatively, though, an `emptyCast` var or the like could be used. `empty()` is used only twice, but it's probably easier to read as a method.